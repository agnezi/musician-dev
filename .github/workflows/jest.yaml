name: Run Jest Tests

on: pull_request

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Set up Node.js
        uses: actions/setup-node@v4
        with:
          node-version: lts/Alpine

      - name: Install dependencies
        run: npm ci

      - name: Run Jest tests
        run: npm test
