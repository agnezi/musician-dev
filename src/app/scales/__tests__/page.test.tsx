import { fireEvent, render, screen } from '@testing-library/react';
import Page from '@/app/scales/page';
import { toneNow, toneTriggerAttackRelease } from '../../../../jest.setup';

describe('scales page', () => {
  it('renders scales page unchanged', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });

  it('should render component when page loads', () => {
    render(<Page />);

    const heading = screen.getByRole('heading', { level: 1 });

    expect(heading).toBeInTheDocument();
  });

  it('should have a button for each scale', () => {
    render(<Page />);

    const buttons = screen.getAllByRole('button');
    expect(buttons[0]).toHaveAttribute('value', 'C Major');
  });

  it('should call tone.js toneTriggerAttackRelease when button click', () => {
    render(<Page />);

    const buttons = screen.getAllByRole('button');
    fireEvent.click(buttons[0]);
    expect(toneTriggerAttackRelease).toHaveBeenCalledWith(
      'C4',
      '8n',
      1234
    );
  });
});
