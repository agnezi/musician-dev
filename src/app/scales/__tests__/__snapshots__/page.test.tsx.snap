// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scales page renders scales page unchanged 1`] = `
<div>
  <div
    class="page"
  >
    <header>
      <h1>
        Major Natural scales notes
      </h1>
    </header>
    <main
      class="main"
    >
      <ol>
        <li>
          <button
            class="scaleButton"
            value="C Major"
          >
            C Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            C4
             
            -
             
          </span>
          <span>
            D4
             
            -
             
          </span>
          <span>
            E4
             
            -
             
          </span>
          <span>
            F4
             
            -
             
          </span>
          <span>
            G4
             
            -
             
          </span>
          <span>
            A4
             
            -
             
          </span>
          <span>
            B4
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="D Major"
          >
            D Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            D4
             
            -
             
          </span>
          <span>
            E4
             
            -
             
          </span>
          <span>
            F#4
             
            -
             
          </span>
          <span>
            G4
             
            -
             
          </span>
          <span>
            A4
             
            -
             
          </span>
          <span>
            B4
             
            -
             
          </span>
          <span>
            C#5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="E Major"
          >
            E Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            E4
             
            -
             
          </span>
          <span>
            F#4
             
            -
             
          </span>
          <span>
            G#4
             
            -
             
          </span>
          <span>
            A4
             
            -
             
          </span>
          <span>
            B4
             
            -
             
          </span>
          <span>
            C#5
             
            -
             
          </span>
          <span>
            D#5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="F Major"
          >
            F Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            F4
             
            -
             
          </span>
          <span>
            G4
             
            -
             
          </span>
          <span>
            A4
             
            -
             
          </span>
          <span>
            Bb4
             
            -
             
          </span>
          <span>
            C5
             
            -
             
          </span>
          <span>
            D5
             
            -
             
          </span>
          <span>
            E5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="G Major"
          >
            G Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            G4
             
            -
             
          </span>
          <span>
            A4
             
            -
             
          </span>
          <span>
            B4
             
            -
             
          </span>
          <span>
            C5
             
            -
             
          </span>
          <span>
            D5
             
            -
             
          </span>
          <span>
            E5
             
            -
             
          </span>
          <span>
            F#5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="A Major"
          >
            A Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            A4
             
            -
             
          </span>
          <span>
            B4
             
            -
             
          </span>
          <span>
            C#5
             
            -
             
          </span>
          <span>
            D5
             
            -
             
          </span>
          <span>
            E5
             
            -
             
          </span>
          <span>
            F#5
             
            -
             
          </span>
          <span>
            G#5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
        <li>
          <button
            class="scaleButton"
            value="B Major"
          >
            B Major
          </button>
          <div
            class="breakLine"
          >
            <br />
          </div>
          <span>
            B4
             
            -
             
          </span>
          <span>
            C#5
             
            -
             
          </span>
          <span>
            D#5
             
            -
             
          </span>
          <span>
            E5
             
            -
             
          </span>
          <span>
            F#5
             
            -
             
          </span>
          <span>
            G#5
             
            -
             
          </span>
          <span>
            A#5
             
             
          </span>
          <div
            class="breakLine"
          >
            <br />
          </div>
        </li>
      </ol>
    </main>
    <footer>
      <a
        href="/"
      >
        Back to home
      </a>
    </footer>
  </div>
</div>
`;
